pkg/common/plugin/download.go

```diff
@@ -0,0 +1,111 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"time"
+
+	"github.com/deislabs/ratify/pkg/ocispecs"
+	"github.com/deislabs/ratify/pkg/referrerstore/oras/authprovider"
+	"github.com/sirupsen/logrus"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+func DownloadPlugin(name string, source string, pluginBinDir string) error {
+	ctx := context.TODO()
+
+	// Initialize a repository
+	repository, err := remote.NewRepository(source)
+	if err != nil {
+		return err
+	}
+
+	// TODO: move the ORAS auth code into a separate package that's not referrerstore-specific
+	repository.Client = &auth.Client{
+		Client: &http.Client{Timeout: 10 * time.Second, Transport: http.DefaultTransport.(*http.Transport).Clone()},
+		Header: http.Header{
+			"User-Agent": {"ratify"},
+		},
+		Cache: auth.NewCache(),
+		Credential: func(ctx context.Context, registry string) (auth.Credential, error) {
+			authProvider, err := authprovider.CreateAuthProviderFromConfig(nil)
```

Comment:
TODO: what should the config look like for specifying auth for pulling plugins. Should look & feel similar to `oras` store config, - but plugins might (will probably?) come from a different registry than application images

Perhaps the config should read something like

```yaml
source:
  reference: noelfdpo.azurecr.io/sample-plugin:v1
  authProvider:
    name: azureWorkloadIdentity
```

Reply:
Agreed, we need to restructure our auth config. Curious do you think customer will use same identity for plugins /artifacts / certs from keyvault or will they use different clientIds?

Reply:
Late follow-up: I would expect to use the same app identity here, at least in Azure

So if you pulled plugins from a different registry than runtime images: it would still be the one "Ratify" Managed Identity that has AcrPull on both the "plugin registry" and the "normal one"